test/PandoraInterface.cxx

```diff
@@ -318,82 +318,86 @@ void ProcessEvents(const Parameters &parameters, const Pandora *const pPrimaryPa
+		    caloHitParameters.m_hitType = pandora::TPC_3D;
+		    caloHitParameters.m_hitRegion = pandora::SINGLE_REGION;
+		    caloHitParameters.m_layer = 0;
+		    caloHitParameters.m_isInOuterSamplingLayer = false;
+		    caloHitParameters.m_pParentAddress = (void *)(static_cast<uintptr_t>(++hitCounter));
+		    caloHitParameters.m_larTPCVolumeId = 0;
+		    caloHitParameters.m_daughterVolumeId = 0;
+
+		    if (parameters.m_use3D)
+		      PANDORA_THROW_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=,
+					      PandoraApi::CaloHit::Create(*pPrimaryPandora, caloHitParameters, m_larCaloHitFactory));
+
+		    if (parameters.m_useLArTPC)
+		      {
+			// Create U, V and W views assuming x is the common drift coordinate
+			const float x0_cm(voxelPos.GetX());
+			const float y0_cm(voxelPos.GetY());
+			const float z0_cm(voxelPos.GetZ());
+
+			lar_content::LArCaloHitParameters caloHitPars_UView(caloHitParameters);
+			caloHitPars_UView.m_hitType = pandora::TPC_VIEW_U;
+			const float upos_cm(pPrimaryPandora->GetPlugins()->GetLArTransformationPlugin()->YZtoU(y0_cm, z0_cm));
+			caloHitPars_UView.m_positionVector = CartesianVector(x0_cm, 0.f, upos_cm);
+
+			lar_content::LArCaloHitParameters caloHitPars_VView(caloHitParameters);
+			caloHitPars_VView.m_hitType = pandora::TPC_VIEW_V;
+			const float vpos_cm(pPrimaryPandora->GetPlugins()->GetLArTransformationPlugin()->YZtoV(y0_cm, z0_cm));
+			caloHitPars_VView.m_positionVector = CartesianVector(x0_cm, 0.f, vpos_cm);
+
+			lar_content::LArCaloHitParameters caloHitPars_WView(caloHitParameters);
+			caloHitPars_WView.m_hitType = pandora::TPC_VIEW_W;
+			const float wpos_cm(pPrimaryPandora->GetPlugins()->GetLArTransformationPlugin()->YZtoW(y0_cm, z0_cm));
+			caloHitPars_WView.m_positionVector = CartesianVector(x0_cm, 0.f, wpos_cm);
+
+			// Create LArCaloHits for U, V and W views
+			PANDORA_THROW_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=,
+						PandoraApi::CaloHit::Create(*pPrimaryPandora, caloHitPars_UView, m_larCaloHitFactory));
+			PANDORA_THROW_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=,
+						PandoraApi::CaloHit::Create(*pPrimaryPandora, caloHitPars_VView, m_larCaloHitFactory));
+			PANDORA_THROW_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=,
+						PandoraApi::CaloHit::Create(*pPrimaryPandora, caloHitPars_WView, m_larCaloHitFactory));
+		      }
+
+		    // Set calo hit voxel to MCParticle relation using trackID
+		    const int trackID = voxel.m_trackID;
+
+		    // Find the energy fraction: voxelHitE/MCParticleE
+		    float energyFrac(0.f), MCEnergy(0.f);
+		    MCParticleEnergyMap::const_iterator mapIter = MCEnergyMap.find(trackID);
+		    if (mapIter != MCEnergyMap.end())
+		      MCEnergy = mapIter->second;
+
+		    if (MCEnergy > 0.0)
+		      energyFrac = voxelE / MCEnergy;
+
+		    PandoraApi::SetCaloHitToMCParticleRelationship(*pPrimaryPandora, (void *)((intptr_t)hitCounter), (void *)((intptr_t)trackID), energyFrac);
+		}
+		else 
+		{
+		  continue;
```

Comment:
I don't think this `continue` is needed, since the code will loop through the voxels anyway.